### PR TITLE
issue/2390-order-detail-status-design-tweaks 

### LINKED ISSUE #2390: Order detail status card design tweaks
Update status card designs as shown below:

screen-shot-2020-04-08-at-2 16 54-pm

- Timestamp and order number are shown first. Timestamp only displays the date of creation (or time if order created the same day)
- Customer’s name is emphasized, without the order number which appears in the screen’s title anyway

### PR DESCRIPTION
**Pull request by nbradbury**
Closes #2390 - updates the header in order detail to shpw date & order number. If date is today show the time.

### Before

before

### After

after

### After (today)

after-today

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt [hunk 1]
```diff
@@ -16,6 +16,7 @@ import java.util.Locale
 
 object DateUtils {
     val friendlyMonthDayFormat by lazy { SimpleDateFormat("MMM d", Locale.getDefault()) }
+    val friendlyTimeFormat by lazy { SimpleDateFormat("hh:mm a", Locale.getDefault()) }
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt:19]: Recommend removing this (see comments later in class)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt [hunk 2]
```diff
@@ -230,6 +231,44 @@ object DateUtils {
...
+        try {
+            val now = GregorianCalendar.getInstance()
+            val date = GregorianCalendar.getInstance().also {
+                it.time = DateTimeUtils.dateUTCFromIso8601(dateString)
+            }
+
+            return now[Calendar.YEAR] == date[Calendar.YEAR] &&
+                now[Calendar.MONTH] == date[Calendar.MONTH] &&
+                now[Calendar.DATE] == date[Calendar.DATE]
+        } catch (e: Exception) {
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt:250]: np: recommend using Android's built in functionality for this: `DateUtils.isSameDay(now, date)`

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt [hunk 3]
```diff
@@ -230,6 +231,44 @@ object DateUtils {
...
+     *
+     * @throws IllegalArgumentException if the argument is not a valid date string.
+     */
+    @Throws(IllegalArgumentException::class)
+    fun getTimeString(dateString: String): String {
+        try {
+            val date = GregorianCalendar.getInstance().also {
+                it.time = DateTimeUtils.dateUTCFromIso8601(dateString)
+            }
+            return friendlyTimeFormat.format(date.time)
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt:266]: Should this use `DateFormat.getTimeFormat(context).format(date.time)` instead so it obeys the time formatting for the specific region? For example, if I set my device to Deutsch, this is how it looked in the **previous version:**

English | Deutsch
-- | --
Screenshot_1589232313|Screenshot_1589232273

This is how it looks in the **new version:**

English | Deutsch
-- | --
Screenshot_1589232324|Screenshot_1589232281

Notice the change in the way the time formatting happens.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt (hunk 2 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt:250]: 🤦

**Comment by nbradbury** [comment]
> Left some code comments regarding localization for your review. Other than that, looks good. :)

@AmandaRiu Thanks for the review - corrections made and ready for another round.

**Pull request merged**